src/main/java/hudson/plugins/audit_trail/ElasticSearchAuditLogger.java

```diff
@@ -0,0 +1,460 @@
+import org.kohsuke.stapler.QueryParameter;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardCertificateCredentials;
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
+import com.google.common.collect.Range;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+
+/**
+ * AuditLogger implementation to send audit logs to an Elastic Search server.
+ * Some code take from the Jenkins logstash plugin: https://github.com/jenkinsci/logstash-plugin
+ *
+ * Default values are set in <code>/src/main/resources/hudson/plugins/audit_trail/ElasticSearchAuditLogger/config.jelly</code>
+ *
+ * @author <a href="mailto:alexander.russell@sap.com">Alex Russell</a>
+ */
+public class ElasticSearchAuditLogger extends AuditLogger {
+
+    private String url;
+    private String usernamePasswordCredentialsId;
+    private String clientCertificateCredentialsId;
+    private boolean skipCertificateValidation = false;
+
+    transient ElasticSearchSender elasticSearchSender;
+
+    protected static final Logger LOGGER = Logger.getLogger(ElasticSearchAuditLogger.class.getName());
+    private static final FastDateFormat DATE_FORMATTER = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ssZ");
+
+    @DataBoundConstructor
+    public ElasticSearchAuditLogger(String url, boolean skipCertificateValidation) {
+        this.url = url;
+        this.skipCertificateValidation = skipCertificateValidation;
+    }
+
+    @Override
+    public void log(String event) {
+        if (elasticSearchSender == null) {
+            // Create the sender because it might not have been created when Jenkins started
+            // The reason for it not being created seems to be because the credentials have not been loaded yet
+            configure();
+            if (elasticSearchSender == null) {
+                LOGGER.log(Level.FINER, "skip log {0}, elasticSearchSender not configured", event);
+                return;
+            }
+        }
+        LOGGER.log(Level.FINER, "Send audit message \"{0}\" to Elastic Search server {1}", new Object[]{event, elasticSearchSender.getUrl()});
+        try {
+            elasticSearchSender.sendMessage(event);
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Exception sending audit message to Elastic Search server " + elasticSearchSender.toString(), e);
+            LOGGER.warning(event);
```

Comment:
When the `event` is null, this kind of log become just "WARNING: null" and that's impossible to track back to where it comes from. Generally it's not adviced to put only a single variable as log message, especially when it could be null as there is no `@Nonnull` or similar annotation.

Recommendation, as the previous log message is already a warning, just merge the two.

(Gut feeling, not tested the null possibility)

Reply:
I'm not sure why wrote that as two log messages.  I don't think it is possible for the event to be null so I have merged the log messages into a single one.  Thanks